### PR TITLE
Getting rid of StatusAPI

### PR DESCRIPTION
**Pull request by adriacidre**
The current implementation of StatusAPI acts as a proxy to StatusBackend, only adding some functions to interact with Backend and its internals.

This first iteration removes all bridged functions and instead opens a `Backend()` function to provide access to the backend itself.

### Important changes:
- [x] Almost all public StatusAPI functions have been removed
- [x] A new `Backend()` function has been added to StatusAPI to directly interact with StatusBackend

### Next steps:
Next step is to fully remove StatusAPI by using a StatusBackend instance instead.

### CODE REVIEW

File: lib/utils.go [hunk 1]
```diff
@@ -14,3 +14,13 @@ func ParseJSONArray(items string) ([]string, error) {
 
 	return parsedItems, nil
 }
+
+func runAsync(f func() error) <-chan error {
```
Comment (adambabik) [inline lib/utils.go]: I would keep it in `api` package. It can be imported into `lib` and we avoid duplication.

**Review by adambabik — COMMENTED** [review]
Maybe we can get rid of `StatusAPI` completely in this PR? I guess the changes won't be much different. Instead of `statusAPI.Backend().CallRPC(C.GoString(inputJSON))` it will be `backend.CallRPC(C.GoString(inputJSON))` and in `lib/main.go` we will change `var statusAPI = api.NewStatusAPI()` to `var backend = api.NewStatusBackend()`

**Comment by adriacidre** [comment]
I can try, wanted to keep the PR small + i had the impression it could grow a lot mainly because of api references on tests, but yeah, happy to remove it here :)

File: t/e2e/api/api_test.go [hunk 2]
```diff
@@ -77,19 +77,19 @@ func (s *APITestSuite) TestRaceConditions() {
 	var funcsToTest = []func(*params.NodeConfig){
 		func(config *params.NodeConfig) {
 			log.Info("StartNodeAsync()")
```
Comment (pedropombeiro) [inline t/e2e/api/api_test.go]: What do we want to do with these log entries? `StartNodeAsync()` no longer exists, and we're making 2 log entries one after another, which seems redundant (maybe there's a reason why one is using `log.Info` and the other `s.T().Logf`).

**Review by pedropombeiro — APPROVED** [review]
LGTM, once runAsync comment from Adam is addressed. Left just a question regarding logging.

File: t/e2e/api/api_test.go (hunk 2 above)
Comment (adriacidre) [reply t/e2e/api/api_test.go]: @PombeirP what do you think about a single log entry just before the runAsync call like:
```
s.T().Logf("Async call to StartNode() for network: %d", config.NetworkID)
api.RunAsync(func() error { return s.backend.StartNode(config) })
```

**Comment by adriacidre** [comment]
@adambabik completely deleted any StatusAPI as you mentioned, I guess at some point we should discuss if `api` package name still makes sense

**Comment by adambabik** [comment]
@adriacidre awesome job! `+115 −349` 👏 

>  I guess at some point we should discuss if api package name still makes sense

I totally agree. It probably makes sense to create a follow-up PR which renames it to `backend` or something else if you have a better idea.

File: t/e2e/api/api_test.go (hunk 2 above)
Comment (pedropombeiro) [reply t/e2e/api/api_test.go]: @adriacidre sounds good :+1:

**Pull request merged**